lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -813,6 +818,56 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
             , expectListField 2
                 (#metrics . #nonMyopicMemberRewards) (`shouldBe` Quantity 0)
             ]
+
+    it "STAKE_POOLS_GARBAGE_COLLECTION_01 - \
+        \retired pools are garbage collected on schedule and not before" $
+        \ctx -> do
+            -- First wait until a few garbage collection cycles have completed.
+            --
+            -- If this test case is run in isolation, this initial waiting
+            -- stage will require a few minutes to complete. However, this
+            -- initial stage is important as it allows the test case to be
+            -- run in isolation, if necessary.
+            --
+            -- If this test case is run at the end of a long integration test
+            -- run, it should run very quickly without much additional delay.
+            --
+            forM_ [1 .. 8] $ \epochNo -> do
```

Comment:
Why 8 and not 5? :sweat_smile:

Reply:
> Why 8 and not 5? sweat_smile

In reality, it could be any number, so long as it's greater than 3. I'll add a comment explaining this choice. :+1:

Reply:
@piotr-iohk Fixed in dfceeb583650284d5cc679e140acafa5d1e76c8c.